app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -214,7 +214,10 @@ def validate(self, data):
                 current_status = getattr(self.instance, status_field)
                 is_locked = current_status not in restricted_statuses
                 is_modifying_other_fields = set(data.keys()) != {status_field}
-                if is_locked and is_modifying_other_fields:
+                is_preview_transition = data == {
+                    "publish_status": NimbusExperiment.PublishStatus.APPROVED
+                }
+                if is_locked and is_modifying_other_fields and not is_preview_transition:
```

Comment:
FWIW, I don't think we want to hard-code an exception for this specific publish_status. We'll very soon (in #4800) want this to also allow `PublishStatus.REVIEW`

This is what I tried locally, treating both `status` and `publish_status` as "unlocked" whenever other fields are restricted:

```diff
diff --git a/app/experimenter/experiments/api/v5/serializers.py b/app/experimenter/experiments/api/v5/serializers.py
index d0a6f7fd..416ba41a 100644
--- a/app/experimenter/experiments/api/v5/serializers.py
+++ b/app/experimenter/experiments/api/v5/serializers.py
@@ -209,11 +209,13 @@ class NimbusStatusValidationMixin:
             "publish_status": NimbusConstants.PUBLISH_STATUS_ALLOWS_UPDATE,
         }
 
+        unlocked_fields = set(restrictive_statuses.keys())
+
         if self.instance:
             for status_field, restricted_statuses in restrictive_statuses.items():
                 current_status = getattr(self.instance, status_field)
                 is_locked = current_status not in restricted_statuses
-                is_modifying_other_fields = set(data.keys()) != {status_field}
+                is_modifying_other_fields = not set(data.keys()).issubset(unlocked_fields)
                 if is_locked and is_modifying_other_fields:
                     raise serializers.ValidationError(
                         {
diff --git a/app/experimenter/experiments/tests/api/v5/test_serializers.py b/app/experimenter/experiments/tests/api/v5/test_serializers.py
index f330c5a8..64e06af3 100644
--- a/app/experimenter/experiments/tests/api/v5/test_serializers.py
+++ b/app/experimenter/experiments/tests/api/v5/test_serializers.py
@@ -1453,3 +1453,18 @@ class TestNimbusStatusValidationMixin(TestCase):
                 "publish_status can be changed."
             ),
         )
+
+    def test_update_publish_status_to_approved_from_preview_status(self):
+        experiment = NimbusExperimentFactory.create(
+            status=NimbusExperiment.Status.PREVIEW,
+            publish_status=NimbusExperiment.PublishStatus.IDLE
+        )
+        serializer = NimbusExperimentSerializer(
+            experiment,
+            data={
+                "id": experiment.id,
+                "publish_status": NimbusExperiment.PublishStatus.APPROVED,
+            },
+            context={"user": self.user},
+        )
+        self.assertTrue(serializer.is_valid())
```